### PR TITLE
Add citation slot to extractor schema

### LINKED ISSUE #10: Add extractor attributes for citation and license info
For citation, we can adopt the CFF/CSL standard for code citation: https://citation-file-format.github.io/ (others are available).

For licenses, we can accept an spdx license identifier and a link to a specific license file https://spdx.org/licenses/.

From discussion with @PeterKraus

### PR DESCRIPTION
**Pull request by ml-evs**
This PR adds a citation slot to extractors based on Dublin Core.

I've asked for advice on the LinkML/OBO slack about the best way of doing this (without needing to inline all these fields).

Also fixed some namespaces for DC which is actually split into terms, elements and types, and tagged the schemas as v0.1.0 (can do a GH release after this PR).

Closes #10

### CODE REVIEW

**Review by PeterKraus — APPROVED** [review]
Nice. Thanks for providing an example.

**Comment by ml-evs** [comment]
I just need to fix the CI before this gets merged, will get around to it later. Will then base the yadg registry entry on this example.

**Comment by ml-evs** [comment]
Encountered some slightly weird behavior with the field name "identifier" and also "id", so replaced it with "uri" for now. If you have `id` or `identifier` it seems to treat these as primary keys in a mapping leaving no syntax available for you to make an array as expected by the schema.

I'll merge this once tests pass without further review but we may want to revisit this.

**Pull request merged**
